test/PositionToken.ts

```diff
@@ -1,15 +1,116 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
 import { expect } from "chai";
+import { ContractFactory } from "ethers";
 import { ethers } from "hardhat";
+import { PositionToken } from "../types/contracts";
+import { buildContractTestContext, ContractTestContext } from "./ContractTestContext";
 
-describe("PositionToken test suit", async function () {
-    let positionToken;
+describe("PositionToken test suit", function () {
+    let r: ContractTestContext;
+    let ptContract: ContractFactory;
 
-    beforeEach(async () => {
-        const ptContract = await ethers.getContractFactory("PositionToken");
-        positionToken = await ptContract.deploy();
+    describe("Roles and initialization", async function () {
+        before(async () => {
+            r = await buildContractTestContext();
+            ptContract = await ethers.getContractFactory("PositionToken");
+        });
+
+        it("Should be built properly by ContractTestContextt", async function () {
+            expect(r.positionToken).to.not.be.undefined;
+        });
+
+        it("Should not allow non admin to init", async function () {
+            const addr1PositionToken = await ptContract.deploy(r.addr1.address) as PositionToken;
+            const initPromise = addr1PositionToken.init(r.addr2.address);
```

Comment:
Just to make surer I understand the rational behind this structure - you dont want to `await` on a method thats going to be reverted for "catching" the revert

Reply:
Sort of, this is basically just to make less code on one line for readability. It is equivalent to just putting the init inside the expect below. The `addr1PositionToken.init` returns a promise and we can choose to await it whenever we want.